.github/workflows/downstream_tests.yml

```diff
@@ -1,9 +1,8 @@
 name: Test Downstream Libraries
 
-on:
-  pull_request:
-  push:
-    branches: [main]
+on: 
+  workflow_call:
+  workflow_dispatch:
```

Comment:
not sure if this is what I meant, but now this can be triggered:
- by other workflows (i.e. in the 1st step of the publish workflow [here](https://github.com/narwhals-dev/narwhals/pull/1228/files#diff-5bdab82401d84fafcce36d5d8c3d9ba4c3682afec172b137defae42bb5f27d43R11)
- manually